### PR TITLE
YLP-1217 Add patient account type during signup

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

File: packages/yourloops/pages/signup/signup-account-form.tsx [hunk 1]
```diff
@@ -53,11 +54,18 @@ interface Errors {
   confirmNewPassword: boolean;
 }
 
+const useStyles = makeStyles((theme: Theme) => ({
+  BackButton: {
```
Comment (TimCanu) [inline packages/yourloops/pages/signup/signup-account-form.tsx:58]: I think that we use camelCase for class name

File: packages/yourloops/pages/signup/signup-account-selector.tsx [hunk 2]
```diff
@@ -65,6 +65,9 @@ const useStyles = makeStyles((theme: Theme) => ({
     marginLeft: theme.spacing(1),
     marginRight: theme.spacing(1),
   },
+  BackButton: {
```
Comment (TimCanu) [inline packages/yourloops/pages/signup/signup-account-selector.tsx:68]: camelCase

File: packages/yourloops/pages/signup/signup-consent.tsx [hunk 3]
```diff
@@ -30,14 +30,22 @@ import { useTranslation } from "react-i18next";
...
 import Button from "@material-ui/core/Button";
+import { makeStyles, Theme } from "@material-ui/core/styles";
 
 import metrics from "../../lib/metrics";
 import { ConsentForm } from "../../components/consents";
 import { useSignUpFormState, FormValuesType } from "./signup-formstate-context";
 import SignUpFormProps from "./signup-form-props";
 
+const useStyles = makeStyles((theme: Theme) => ({
+  BackButton: {
```
Comment (TimCanu) [inline packages/yourloops/pages/signup/signup-consent.tsx:41]: camelCase

File: packages/yourloops/pages/signup/signup-profile-form.tsx [hunk 4]
```diff
@@ -61,6 +61,9 @@ const formStyle = makeStyles((theme: Theme) => {
       marginLeft: theme.spacing(1),
       marginRight: theme.spacing(1),
     },
+    BackButton: {
```
Comment (TimCanu) [inline packages/yourloops/pages/signup/signup-profile-form.tsx:64]: :camel:

File: packages/yourloops/test/pages/stepper/signup-account-selector.test.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  const nextButtonDisabled = (wrapper: ReactWrapper) => {
+    expect(wrapper.find("button#button-signup-steppers-next").prop("disabled")).to.be.true;
+  };
+
+  const nextButtonEnabled = (wrapper: ReactWrapper) => {
+    expect(wrapper.find("button#button-signup-steppers-next").prop("disabled")).to.be.false;
+  };
+
+  it("should disable next button when nothing is selected", async () => {
+    const wrapper = await mount(signupForm());
```
Comment (TimCanu) [inline packages/yourloops/test/pages/stepper/signup-account-selector.test.tsx:55]: opt: Might be possible to into a beforeEach: 
`move const wrapper = await mount(signupForm());`
and in an afterEach:
`wrapper.unmount();`
But it is really not necessary and might not be compatible with future tests :man_teacher:

**Review by TimCanu — CHANGES_REQUESTED** [review]

File: packages/yourloops/models/shoreline.ts [hunk 6]
```diff
@@ -48,6 +48,7 @@ enum UserRoles {
   hcp = "hcp",
   caregiver = "caregiver",
   patient = "patient",
+  empty = "",
```
Comment (yvon-dblg) [inline packages/yourloops/models/shoreline.ts:51]: I would not have changed this definition just for the signup form.
This "empty" should be limited to signup form only.

Maybe try to change `SignupUser` (packages/yourloops/lib/auth/models.ts) to:
```
export interface SignupUser {
[...]
  accountRole?: UserRoles;
[...]
}
```
and remove from `initialState`: accountRole: UserRoles.caregiver

**Review by yvon-dblg — CHANGES_REQUESTED** [review]

File: packages/yourloops/test/pages/stepper/signup-account-selector.test.tsx (hunk 5 above)
Comment (yvon-dblg) [reply packages/yourloops/test/pages/stepper/signup-account-selector.test.tsx:55]: For info:
The unmount() is necessary if you load your component to a DOM element attached to the document.
like:
```javascript
const div = document.createElement("div");
document.body.appendElement(div);
const wrapper = await mount(signupForm(), div);
```

Comment (yanngit) [reply packages/yourloops/test/pages/stepper/signup-account-selector.test.tsx:55]: done

File: packages/yourloops/models/shoreline.ts (hunk 6 above)
Comment (yanngit) [reply packages/yourloops/models/shoreline.ts:51]: you're right, it was easier for me but maybe not a good solution ... I'm refactoring :)

File: packages/yourloops/lib/auth/models.ts [hunk 7]
```diff
@@ -47,7 +47,7 @@ export interface UpdateUser {
 export interface SignupUser {
   accountUsername: string;
   accountPassword: string;
-  accountRole: UserRoles;
+  accountRole: UserRoles|undefined;
```
Comment (yvon-dblg) [inline packages/yourloops/lib/auth/models.ts:50]:
```suggestion
  accountRole?: UserRoles;
```

File: packages/yourloops/pages/signup/signup-formstate-context.tsx [hunk 8]
```diff
@@ -28,10 +28,11 @@
 import React from "react";
 import _ from "lodash";
 
+import { HcpProfession } from "../../models/hcp-profession";
 import { UserRoles } from "../../models/shoreline";
+
```
Comment (TimCanu) [inline packages/yourloops/pages/signup/signup-formstate-context.tsx:33]: Is there a reason for this extra space? Like, is it part of the imports rule?

**Review by TimCanu — APPROVED** [review]

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**